docs/api_pins.rst

```diff
@@ -11,68 +11,68 @@ are concerned with. However, some users may wish to take advantage of the
 
-When you construct a device, you pass in a GPIO pin number. However, what the
-library actually expects is a :class:`Pin` implementation. If it finds a simple
-integer number instead, it uses one of the following classes to provide the
-:class:`Pin` implementation (classes are listed in favoured order):
-
-1. :class:`gpiozero.pins.rpigpio.RPiGPIOPin`
-
-2. :class:`gpiozero.pins.rpio.RPIOPin`
-
-3. :class:`gpiozero.pins.pigpiod.PiGPIOPin`
-
-4. :class:`gpiozero.pins.native.NativePin`
-
-You can change the default pin implementation by over-writing the
-``pin_factory`` global in the ``devices`` module like so::
-
-    from gpiozero.pins.native import NativePin
-    import gpiozero.devices
-    # Force the default pin implementation to be NativePin
-    gpiozero.devices.pin_factory = NativePin
+When you construct a device, you pass in a pin specification. However, what the
+library actually expects is a :class:`Pin` implementation. If it finds anything
+else, it uses the existing ``Device._pin_factory`` to construct a :class:`Pin`
+implementation based on the specification.
+
+Changing the pin factory
+========================
+
+The default pin factory can be replaced by specifying a value for the
+``GPIOZERO_PIN_FACTORY`` environment variable. For example:
+
+.. code-block:: console
+
+    pi@raspberrypi $ GPIOZERO_PIN_FACTORY=native python
+    Python 3.4.2 (default, Oct 19 2014, 13:31:11)
+    [GCC 4.9.1] on linux
+    Type "help", "copyright", "credits" or "license" for more information.
+    >>> import gpiozero
+    >>> gpiozero.Device._pin_factory
+    <gpiozero.pins.native.NativeFactory object at 0x762c26b0>
+
+The following values, and the corresponding :class:`Factory` and :class:`Pin`
+classes are listed in the table below. Factories are listed in the order that
+they are tried by default.
+
++---------+-----------------------------------------------+-------------------------------------------+
+| Name    | Factory class                                 | Pin class                                 |
++=========+===============================================+===========================================+
+| rpigpio | :class:`gpiozero.pins.rpigpio.RPiGPIOFactory` | :class:`gpiozero.pins.rpigpio.RPiGPIOPin` |
++---------+-----------------------------------------------+-------------------------------------------+
+| rpio    | :class:`gpiozero.pins.rpio.RPIOFactory`       | :class:`gpiozero.pins.rpio.RPIOPin`       |
++---------+-----------------------------------------------+-------------------------------------------+
+| pigpio  | :class:`gpiozero.pins.pigpiod.PiGPIOFactory`  | :class:`gpiozero.pins.pigpiod.PiGPIOPin`  |
++---------+-----------------------------------------------+-------------------------------------------+
+| native  | :class:`gpiozero.pins.native.NativeFactory`   | :class:`gpiozero.pins.native.NativePin`   |
++---------+-----------------------------------------------+-------------------------------------------+
+
```

Comment:
Do `MockFactory` and `MockPWMFactory` need to be documented anywhere? Or are these strictly "for internal testing only" and therefore shouldn't be publicly documented?
If only there was a developer documentation page.... ;-)